src/PointEvalHandler.jl

```diff
@@ -202,6 +209,31 @@ function _get_node_cell_map(grid::AbstractGrid)
     return cell_dicts
 end
 
+function _get_node_cell_map(grid::AbstractGrid, subdomains::Vector{OrderedSet{Int}})
+    @inline function _check_contained_subdomain(i, subdomains)
+        for subdomain in subdomains
+            i ∈ subdomain && return true
+        end
+        return false
+    end
+    cells = getcells(grid)
+    C = eltype(cells) # possibly abstract
+    cell_dicts = Dict{Type{<:C}, Dict{Int, Vector{Int}}}()
+    ctypes = Set{Type{<:C}}(typeof(c) for c in cells)
+    for ctype in ctypes
+        cell_dict = cell_dicts[ctype] = Dict{Int, Vector{Int}}()
+        for (cellidx, cell) in enumerate(cells)
+            cell isa ctype || continue
+            _check_contained_subdomain(cellidx, subdomains) || continue
```

Comment:
Isn't this enough, or was a performance optimization required?
```suggestion
            any(set -> cellidx in set, subdomains) || continue
```

But if following other comments, would of course also remove the `any`...